controllers/onload_controller.go

```diff
@@ -108,6 +122,84 @@ func (r *OnloadReconciler) Reconcile(
 	return ctrl.Result{}, nil
 }
 
+func getLabelName(onload onloadv1alpha1.Onload) string {
+	return "onload.amd.com/" + onload.Namespace + "." + onload.Name
+}
+
+func (r *OnloadReconciler) deleteLabels(ctx context.Context, namespacedName types.NamespacedName) error {
+	log := log.FromContext(ctx)
+
+	nodes := corev1.NodeList{}
+	err := r.List(ctx, &nodes)
+	if err != nil {
+		log.Error(err, "Failed to list nodes")
+		return err
+	}
+
+	for _, node := range nodes.Items {
+		nodeCopy := node.DeepCopy()
+		labels := node.GetObjectMeta().GetLabels()
+		for k := range labels {
+			if strings.HasPrefix(k, "onload.amd.com") {
+				delete(labels, k)
+			}
+		}
+		err := r.Patch(ctx, &node, client.MergeFrom(nodeCopy))
```

Comment:
Does `r.List()` return all nodes in the cluster? There may be thousands of nodes so we should only touch what we must: `r.Patch()` should only run if above `.hasPrefix()` matched, and potentially also only do DeepCopy if match, depending on how slow DeepCopy is.

Reply:
yeah, I think there is a bug here anyway so I will update. This should have a selector to only get the nodes with that match the selector (similar to [labelNodes](https://github.com/Xilinx-CNS/kubernetes-onload/pull/52/files/f2ab2a8a54200d0948f2f1f2bf690d1e91c34b2d#diff-24b008afa2ecacf8904ab1d398c873e662b7bb8ac5ee5bf343abf7d272668c81R160-R161))